website/docs/flink_configuration.md

```diff
@@ -3,115 +3,179 @@ title: Flink Setup
-   desired memory size `write.task.max.size`. For example, taskManager has `4GB` of memory running two streamWriteFunction, so each write task
-   can be allocated with `2GB` memory. Please reserve some buffers because the network buffer and other types of tasks on taskManager (such as bucketAssignFunction) will also consume memory.
-4. Pay attention to the memory changes of compaction. `compaction.max_memory` controls the maximum memory that each task can be used when compaction tasks read
-   logs. `compaction.tasks` controls the parallelism of compaction tasks.
-
-### COW
-
-1. [Setting Flink state backend to `rocksdb`](#checkpoint) (the default `in memory` state backend is very memory intensive).
-2. Increase both `write.task.max.size` and `write.merge.max_memory` (`1024MB` and `100MB` by default, adjust to `2014MB` and `1024MB`).
-3. Pay attention to the memory allocated to each write task by taskManager to ensure that each write task can be allocated to the
-   desired memory size `write.task.max.size`. For example, taskManager has `4GB` of memory running two write tasks, so each write task
-   can be allocated with `2GB` memory. Please reserve some buffers because the network buffer and other types of tasks on taskManager (such as `BucketAssignFunction`) will also consume memory.
-
-
-## Write Rate Limit
+## Configure Flink Execution Modes
+You can configure the execution mode via the `execution.runtime-mode` setting. There are three possible modes:
 
-In the existing data synchronization, `snapshot data` and `incremental data` are send to kafka first, and then streaming write
-to Hudi by Flink. Because the direct consumption of `snapshot data` will lead to problems such as high throughput and serious
-disorder (writing partition randomly), which will lead to write performance degradation and throughput glitches. At this time,
-the `write.rate.limit` option can be turned on to ensure smooth writing.
-
-### Options
-
-|  Option Name  | Required | Default | Remarks |
-|  -----------  | -------  | ------- | ------- |
-| `write.rate.limit` | `false` | `0` | Turn off by default |
\ No newline at end of file
+- **STREAMING**: The classic DataStream execution mode (default)
+- **BATCH**: Batch-style execution on the DataStream API
+- **AUTOMATIC**: Let the system decide based on the boundedness of the sources
+
+You can configured the execution mode via the command line:
+
+```sh
+$ bin/flink run -Dexecution.runtime-mode=BATCH <jarFile>
+
+```
+
+Separately, you can programmatically create and configure the `StreamExecutionEnvironment`, a Flink programming API. This execution environment is how all data pipelines are created and maintained.
+
+Below, you can configure the execution mode via the command line. The execution mode is used to configure a streaming, batch and automatic runtime mode. By default, the `streaming` execution mode is enabled. 
```

Comment:
Rephrase to -> "Below is an example of how to set the `BATCH` mode programmatically."

Reply:
You can configure the execution mode programmatically. Below is an example of how to set the `BATCH` mode. By default, the `streaming` execution mode is enabled. 

cc @bhasudha wdyt?